src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,104 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private int capacity = DEFAULT_CAPACITY;
+    private int threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    private Node<K, V>[] table;
+    private int size = 0;
 
     @Override
     public void put(K key, V value) {
+        if (table == null) {
+            table = new Node[capacity];
+        } else if (size == threshold) {
+            resize();
+        }
+        int keyHash = (key == null) ? 0 : key.hashCode();
+        int indexOfBucket = Math.abs(keyHash % capacity);
+        if (table[indexOfBucket] == null) {
+            table[indexOfBucket] = new Node<>(keyHash, key, value, null);
+            size++;
+        } else {
+            putIfBucketByIndexExist(keyHash, key, value, indexOfBucket);
+        }
+    }
 
+    private void putIfBucketByIndexExist(int keyHash, K key, V value, int indexOfBucket) {
+        Node<K, V> currentNode = table[indexOfBucket];
+        while (currentNode.next != null) {
+            if (currentNode.key != null && currentNode.key.equals(key) || currentNode.key == key) {
+                currentNode.value = value;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+        if ((currentNode.key != null && currentNode.key.equals(key)) || currentNode.key == key) {
+            currentNode.value = value;
+        } else {
+            currentNode.next = new Node<>(keyHash, key, value, null);
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (table != null) {
+            for (int i = 0; i < table.length; i++) {
```

Comment:
you dont have to iterate over the whole table, just get index by the hashcode of key